main/modes/timer/modeTimer.c

```diff
@@ -0,0 +1,286 @@
+    {
+        freeWsg(&timerData->hourglassFrames[i]);
+    }
+
+    freeFont(&timerData->textFont);
+    freeFont(&timerData->numberFont);
+
+    free(timerData);
+    timerData = NULL;
+}
+
+static void timerMainLoop(int64_t elapsedUs)
+{
+    int64_t now = esp_timer_get_time();
+    if (timerData->timerState == RUNNING
+        && (!timerData->stopwatch
+            && timerData->accumulatedDuration + (now - timerData->startTime) >= timerData->countdownTime))
+    {
+        timerData->accumulatedDuration = timerData->countdownTime;
+        timerData->timerState          = EXPIRED;
+    }
+
+    buttonEvt_t evt;
+    while (checkButtonQueueWrapper(&evt))
+    {
+        if (evt.down && evt.button == PB_A)
+        {
+            // A pressed, pause/unpause/start
+            switch (timerData->timerState)
+            {
+                case STOPPED:
+                    timerData->accumulatedDuration = 0;
+                    // Intentional fall-through
+                case PAUSED:
+                {
+                    timerData->startTime  = now;
+                    timerData->timerState = RUNNING;
+                    break;
+                }
+
+                case RUNNING:
+                {
+                    timerData->accumulatedDuration += (now - timerData->startTime);
+                    timerData->timerState = PAUSED;
+                    break;
+                }
+
+                case EXPIRED:
+                {
+                    timerData->timerState = STOPPED;
+                    break;
+                }
+            }
+        }
+        else if (evt.down && evt.button == PB_B)
+        {
+            // B pressed, pause/stop/reset
+            // TODO
+            switch (timerData->timerState)
+            {
+                case PAUSED:
+                {
+                    timerData->timerState          = STOPPED;
+                    timerData->accumulatedDuration = 0;
+                    break;
+                }
+
+                case RUNNING:
+                {
+                    timerData->accumulatedDuration += (now - timerData->startTime);
+                    timerData->timerState = PAUSED;
+                    break;
+                }
+
+                case STOPPED:
+                case EXPIRED:
+                    break;
+            }
+        }
+        else if (evt.down && (evt.button == PB_LEFT || evt.button == PB_RIGHT))
+        {
+            if (timerData->timerState == STOPPED)
+            {
+                timerData->stopwatch = !timerData->stopwatch;
+            }
+        }
+        else if (evt.down && (evt.button == PB_UP || evt.button == PB_DOWN))
+        {
+            if (timerData->timerState == STOPPED)
+            {
+                if (!timerData->stopwatch)
+                {
+                    timerData->countdownTime += (evt.button == PB_UP) ? 30000000 : -30000000;
+                }
+            }
+        }
```

Comment:
Ah, you should only draw helper text for switching the modes when the timer is `STOPPED` it seems,